### DISCUSSION TITLE
IOS App Tracking Transparency for Admob

### DISCUSSION THREAD

**Discussion by abdullahskartal** [invertase/react-native-firebase#5340] [Q&A] [answered]
I have an app where I want to show personalized ads so I want to use ATT for IOS 14+.  After some research I found the [RN-Tracking-Trasparency](https://github.com/mrousavy/react-native-tracking-transparency) package but I'm confused because of Admob.

As I understand it if user does not allow for ATT I should disable tracking Ad ID for Admob. On the contrary if user allow for ATT I can show personalize ads for user so I should enable tracking Ad ID feature (or default enable so i shouldn't do anything) How can I do this depending on the user given permission condition?

I saw the [section](https://rnfirebase.io/analytics/usage#disable-ad-id-usage-on-ios) in the documentation, but I don't think it's conditional.

Thanks in advance.

**Comment by mikehardy** [comment, marked as answer]
Use react-native-permissions for ATT, I think.
 I can't speak to integration with AdMob, they removed it from the upstream firebase SDKs so we don't relate to it directly at the moment.

If you want to use AdMob, you *must* integrate ATT though, as it uses IDFA symbols from iOS.
There are lots of toggles for whether data is collected by default on startup that are available in firebase.json you can see the schema here (v11.5.0 version from here, where we still had a firebase admob module) https://github.com/invertase/react-native-firebase/blob/v11.5.0/packages/app/firebase-schema.json

**Reply by abdullahskartal** [reply]
I feel like I haven't seen a value in the schema that fully meets ATT. I am currently using v11.5.0 version so should I use "admob_delay_app_measurement_init" for ATT toggle?

**Reply by mikehardy** [reply]
ATT must be implemented if you include the IDFA symbols at all
Then you must delay all use of the IDFA symbols (using that JSON key, yes plus also analytics data collection I think) until you request (and receive!) ATT permission. I think you're on the right path.

**Reply by abdullahskartal** [reply]
Thank you!
